### ISSUE TITLE
Locale extraction with Cldr.Plug.SetLocale

### ISSUE THREAD

**Issue by BrummbQ** [kipcole9/cldr#33]
The locale parameter extraction in this plug puzzles me. See i.e. this route:
```
scope "/:locale", MyApp do
   #stuff
end
```
Which configuration should i use? `SetLocale, apps: [:gettext, :cldr], from: [:url]` isn't working. Looking at the code I don't understand the parameter url_params. conn.url_params doesn't even exist here

Have a nice day!

**Comment by kipcole9** [comment]
Thanks for the report.  Definitely incorrect to be using `url_params` and I have corrected this now on master to be the correct `path_params`.  That will also mean it will be `SetLocale, apps: [:gettext, :cldr], from: :path`

I believe in Phoenix the right way to invoke this (with the corrected `:path` option) is to include `Cldr.Plug.SetLocale` into a pipeline.  Using the Phoenix docs as a guide it would be:

```
defmodule MyAppWeb.Router do
  use Phoenix.Router

  pipeline :localisation do
    plug Cldr.Plug.SetLocale, from: :path
  end

  scope "/" do
    pipe_through :localisation

    # controller matching
  end
end
```
I'm not strong on Phoenix (but Im more than happy to collaborate to get Cldr rock-solid in support of it).  So for now may I ask you to try:

1. Install Cldr from GitHub with `{:ex_cldr, git: "https://github.com/kipcole9/cldr"}` so that the correct name of `conn.path_params` is being used

2. Try using a `pipeline` like the example above which will invoke the `Cldr.Plug.SetLocale` plug in the context of your scope.

In parallel I will work out how to put some tests together for this.  Apologies for the inconvenience but I'll work on this over the next 24 hours.

**Comment by kipcole9** [comment]
Fixed in [version 1.3.1](https://hex.pm/packages/ex_cldr) which is now on hex.  I have created a couple of test cases in `test/plug/plug_router_test.exs` to validate that the `pipeline` approach is the way to go.  I fixed a couple of bugs as well.

Sorry for the inconvenience and hope you find this issue is now resolved.
